src/wazuh_modules/syscollector/sysInfoWin.cpp

```diff
@@ -8,18 +8,280 @@
+    DWORD pageFileUsage() const
+    {
+        return m_pageFileUsage;
+    }
+
+    DWORD virtualSize() const
+    {
+        return m_virtualSize;
+    }
+
+    DWORD sessionId() const
+    {
+        DWORD ret{};
+        if (!ProcessIdToSessionId(m_pId, &ret))
+        {
+            throw std::system_error
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to retrieve session ID from process with PID %lu" + std::to_string(m_pId)
+            };
+        }
+        return ret;
+    }
+
+private:
+    void setProcessTimes()
+    {
+        static const auto s_toSecondsValue { 10000000ULL };
+        FILETIME lpCreationTime{};
+        FILETIME lpExitTime{};
+        FILETIME lpKernelTime{};
+        FILETIME lpUserTime{};
+        if (GetProcessTimes(m_hProcess, &lpCreationTime, &lpExitTime, &lpKernelTime, &lpUserTime))
+        {
+            // Copy the kernel mode filetime high and low parts and convert it to seconds
+            m_kernelModeTime.LowPart = lpKernelTime.dwLowDateTime;
+            m_kernelModeTime.HighPart = lpKernelTime.dwHighDateTime;
+            m_kernelModeTime.QuadPart /= s_toSecondsValue;
+
+            // Copy the user mode filetime high and low parts and convert it to seconds
+            m_userModeTime.LowPart = lpUserTime.dwLowDateTime;
+            m_userModeTime.HighPart = lpUserTime.dwHighDateTime;
+            m_userModeTime.QuadPart /= s_toSecondsValue;
+        }
+        else
+        {
+            throw std::system_error
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to retrieve kernel mode and user mode times from process with PID %lu" + std::to_string(m_pId)
+            };          
+        }
+    }
+
+    void setProcessMemInfo()
+    {
+        PROCESS_MEMORY_COUNTERS pMemCounters{};
+        // Get page file usage and virtual size
+        // Reference: https://stackoverflow.com/a/1986486
+        if (GetProcessMemoryInfo(m_hProcess, &pMemCounters, sizeof(pMemCounters)))
+        {
+            m_pageFileUsage = pMemCounters.PagefileUsage;
+            m_virtualSize   = pMemCounters.WorkingSetSize + pMemCounters.PagefileUsage;
+        }
+        else
+        {
+            throw std::system_error
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to retrieve page file usage from process with PID %lu" + std::to_string(m_pId)
+            };
+        }
+    }
+
+    bool ntPath2Win32Path(char* ntpath, char** outbuf)
+    {
+        int success { 0 };
+
+        if (nullptr != ntpath)
+        {
+            char logicalDrives[OS_MAXSTR] = {0};
+            // Get the total amount of available logical drives
+            // The input size must not include the NULL terminator
+            DWORD res { GetLogicalDriveStrings(OS_MAXSTR - 1, logicalDrives) };
```

Comment:
use a vector of pairs <LogicalDriveStrings, DosDriveStrings>
1- check if there is a match of one of the LocalDriveStrings and the beginning of the path to be transformed.
2- if so, replace the strings.
3- if not, refill the vector of paris and try again.

Reply:
Will do.